app/services/proxy_deployment_service.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+class ProxyDeploymentService
+  delegate :apicast_configuration_driven,
+           :deployable?,
+           :service_mesh_integration?,
+           :provider,
+           :proxy_configs, to: :@proxy
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(proxy)
+    @proxy = proxy
+  end
+
+  def call
```

Comment:
```suggestion
  def call(environment: :sandbox)
    return true unless deployable?
    return deploy_production if environment == :production
```

Not entirely sure, but I'd start with an option and later maybe worth splitting into one separate classes, one per environment, as they seem to share virtually no logic. At least this first service abstracts already from start all the complexity. The refactor in to more specific classes can be done in a next iteration.

@hallelujah , WDYT?

Reply:
I added the parameter in the constructor but I agree there should be a second class

Reply:
I think they share some logic. Having 2 classes might not be necessary but does not do any harm either  👍